python/pyarrow/_dataset.pyx

```diff
@@ -1251,6 +1266,9 @@ cdef class CsvFragmentScanOptions(FragmentScanOptions):
 
     cdef:
         CCsvFragmentScanOptions* csv_options
+        # The encoding field in ReadOptions does not exist in the C++ struct.
+        # We need to store it here and override it when reading read_options
+        ReadOptions read_options_py
```

Comment:
Same here.

Reply:
Changed in [b9982c8](https://github.com/apache/arrow/pull/13820/commits/b9982c8d28d898bbf506135bf53bad629728afef)